text-summarization/run_service.py

```diff
@@ -10,39 +11,44 @@ def main():
     parser = argparse.ArgumentParser(prog="run-snet-services")
     parser.add_argument("--daemon-config-path", help="Path to daemon configuration file", required=False)
     args = parser.parse_args(sys.argv[1:])
-
-    def handle_signal(signum, frame):
-        snetd_p.send_signal(signum)
-        service_p.send_signal(signum)
-        snetd_p.wait()
-        service_p.wait()
-        exit(0)
-
-    signal.signal(signal.SIGTERM, handle_signal)
-    signal.signal(signal.SIGINT, handle_signal)
-
-    root_path = pathlib.Path(__file__).absolute().parent.parent
-    snetd_p = start_snetd(root_path, args.daemon_config_path)
-    service_p = start_service(root_path)
-
-    while True:
-        if snetd_p.poll() is not None:
-            snetd_p = start_snetd(root_path, args.daemon_config_path)
-        if service_p.poll() is not None:
-            service_p = start_service(root_path)
-        time.sleep(5)
+    
+    root_path = pathlib.Path(__file__).absolute().parent
+    all_p = [start_snetd(root_path, args.daemon_config_path), start_service(root_path)]
+    
+    # Continuous checking all subprocess
+    try:
+        while True:
+            for p in all_p:
+                p.poll()
+                if p.returncode and p.returncode != 0:
+                    kill_and_exit(all_p)
+            time.sleep(1)
+    except Exception as e:
+        print(e)
```

Comment:
If we are doing nothing except catching generic exceptions and printing them, I would delete this `try: except` wrapper. This pattern hides the stack trace and is annoying for no real benefit.

Reply:
You're right, removing try-except...